src/test/groovy/graphql/normalized/ExecutableNormalizedOperationToAstCompilerDeferTest.groovy

```diff
@@ -0,0 +1,552 @@
+          }
+        """
+        GraphQLSchema schema = mkSchema(sdl)
+        def tree = createNormalizedTree(schema, query)
+        when:
+        def result = compileToDocumentWithDeferSupport(schema, QUERY, null, tree.topLevelFields, noVariables, tree.normalizedFieldToDeferExecution)
+        def printed = AstPrinter.printAst(new AstSorter().sort(result.document))
+        then:
+        printed == '''{
+  dog {
+    name
+    ... @defer(label: "breed-defer") {
+      breed
+    }
+  }
+}
+'''
+    }
+
+    def "multiple labels on the same field"() {
+        String query = """
+          query q {
+            dog {
+                name
+                ... @defer(label: "breed-defer") {
+                    breed
+                }
+                ... @defer(label: "breed-defer-2") {
+                    breed
+                }
+            }
+          }
+        """
+        GraphQLSchema schema = mkSchema(sdl)
+        def tree = createNormalizedTree(schema, query)
+        when:
+        def result = compileToDocumentWithDeferSupport(schema, QUERY, null, tree.topLevelFields, noVariables, tree.normalizedFieldToDeferExecution)
+        def printed = AstPrinter.printAst(new AstSorter().sort(result.document))
+        then:
+        printed == '''{
+  dog {
+    name
+    ... @defer(label: "breed-defer") {
+      breed
+    }
+    ... @defer(label: "breed-defer-2") {
+      breed
+    }
+  }
+}
+'''
+    }
+
+    def "multiple defers without label on the same field"() {
+        String query = """
+          query q {
+            dog {
+                name
+                ... @defer {
+                    breed
+                }
+                ... @defer {
+                    breed
+                }
+            }
+          }
+        """
+        GraphQLSchema schema = mkSchema(sdl)
+        def tree = createNormalizedTree(schema, query)
+        when:
+        def result = compileToDocumentWithDeferSupport(schema, QUERY, null, tree.topLevelFields, noVariables, tree.normalizedFieldToDeferExecution)
+        def printed = AstPrinter.printAst(new AstSorter().sort(result.document))
+        then:
+        printed == '''{
+  dog {
+    name
+    ... @defer {
+      breed
+    }
+    ... @defer {
+      breed
+    }
+  }
+}
+'''
+    }
+
+    def "field with and without defer"() {
+        String query = """
+          query q {
+            dog {
+                ... @defer {
+                    breed
+                }
+                ... {
+                    breed
+                }
+            }
+          }
+        """
+        GraphQLSchema schema = mkSchema(sdl)
+        def tree = createNormalizedTree(schema, query)
+        when:
+        def result = compileToDocumentWithDeferSupport(schema, QUERY, null, tree.topLevelFields, noVariables, tree.normalizedFieldToDeferExecution)
+        def printed = AstPrinter.printAst(new AstSorter().sort(result.document))
+        then:
+        printed == '''{
+  dog {
```

Comment:
Not a blocker, but we should double-check what is the expected behaviour here with the drivers of the defer/stream spec change.
For reference, the Apollo server execution will block the response until `breed` is return in the initial payload, then it will sent a second, deferred, payload with `breed` again.

Reply:
Based on the direction the working group is taking (https://github.com/graphql/defer-stream-wg/discussions/69) regarding de-duplication, I propose we keep only the non-deferred version of the field. 
I can do it on a follow up PR.